deepchecks/vision/checks/distribution/image_property_drift.py

```diff
@@ -168,17 +180,87 @@ def compute(self, context: Context) -> CheckResult:
                        '</span>'
 
-            displays = [headnote] + [figures[col] for col in columns_order]
-        else:
-            drifts = None
-            displays = []
+            train_samples = df_train.sample(10)
+            test_samples = df_test.sample(10)
+            train_thumbnail_images, *_ = context.train.batch_of_index(*list(train_samples.index))
+            test_thumbnail_images, *_ = context.test.batch_of_index(*list(train_samples.index))
+
+            thumbnails = self._prepare_thumbnails_block(
+                train_properties=train_samples.T,
+                test_properties=test_samples.T,
+                train_images=train_thumbnail_images,
+                test_images=test_thumbnail_images
+            )
+            displays = [headnote] + [figures[col] for col in columns_order] + [thumbnails]
 
         return CheckResult(
             value=drifts,
             display=displays,
             header='Image Property Drift'
         )
 
+    def _prepare_thumbnails_block(
+        self,
+        train_properties: pd.DataFrame,
+        test_properties: pd.DataFrame,
+        train_images: t.Sequence[pilimage.Image],
+        test_images: t.Sequence[pilimage.Image]
+    ) -> str:
+        table_template = dedent("""
+        <div
+            style="
+                overflow-x: auto;
+                display: grid;
+                grid-template-rows: auto 1fr 1fr;
+                grid-template-columns: auto repeat({n_of_images}, 1fr);
+                grid-gap: 1.5rem;
+                justify-items: center;
+                align-items: center;
+                padding: 2rem;
+                width: max-content;">
+            <h4>Image</h4>
+            {images}
+            {properties}
+        </div>
+        """)
+
+        thumbnail_size = self._IMAGE_THUMBNAIL_SIZE
+        tables = []
+
+        for images, properties in ((train_images, train_properties), (test_images, test_properties),):
+            thumbnails = ''.join([
+                prepare_thumbnail(img, size=thumbnail_size)
+                for img in images
+            ])
+
+            properties_rows = []
+            for name, values in properties.iterrows():
+                properties_rows.append(f'<h4>{name}</h4>')
+                for v in values:
+                    properties_rows.append(f'<h4>{format_number(v)}</h4>')
+
+            tables.append(table_template.format(
+                n_of_images=len(images),
+                images=thumbnails,
+                properties=''.join(properties_rows)
+            ))
+
+        train_table, test_table = tables
+
+        template = dedent("""
+        <h4>Train Images</h3>
+        <hr>
+        {train_thumbnails}
+        <h4>Test Images</h4>
+        <hr>
+        {test_thumbnails}
+        """)
+
+        return template.format(
+            train_thumbnails=train_table,
+            test_thumbnails=test_table,
+        )
+
```

Comment:
maybe it's better to have it in a util file? since many checks will have to display thumbnails of images.

Reply:
this particular function is too specific but I decoupled grid creation into a separate function called `prepare_grid`